crates/bevy_ecs/src/component.rs

```diff
@@ -186,7 +186,7 @@ impl std::fmt::Debug for ComponentDescriptor {
 impl ComponentDescriptor {
     // SAFETY: The pointer points to a valid value of type `T` and it is safe to drop this value.
     unsafe fn drop_ptr<T>(x: OwningPtr<'_>) {
```

Comment:
I'm not sure we really need `ComponentDescriptor::drop_ptr` anymore considering you can just do `OwningPtr::drop_as`

Reply:
I actually tried removing this and there was a lifetime mismatch. Using `OwnedPtr::drop_as::<T>` doesn't satisfy `for<'a> unsafe fn...`. Is there a way to make this work?

Reply:
oh... that makes sense I guess, oh well ._.